src/coreclr/jit/emitxarch.cpp

```diff
@@ -7055,12 +7055,170 @@ void emitter::emitIns_S(instruction ins, emitAttr attr, int varx, int offs)
     emitAdjustStackDepthPushPop(ins);
 }
 
+//----------------------------------------------------------------------------------------
+// IsRedundantMov_S_R:
+//    Check if the current `mov` instruction is redundant and can be omitted when dealing with Load/Store from stack.
+//    A `mov` is redundant in following 2 cases:
+//
+//    1. Move that is identical to last instruction emitted.
+//
+//         vmovapd  xmmword ptr [V01 rbp-20H], xmm0  # <-- last instruction
+//         vmovapd  xmmword ptr [V01 rbp-20H], xmm0  # <-- current instruction can be omitted.
+//
+//    2. Opposite Move as that of last instruction emitted.
```

Comment:
Isn't this case `*_R_S` rather than `*_S_R`?

Reply:
It covers both R_S and S_R

Reply:
Maybe we could call this `IsRedundantStackMov` or similar?

Someone on the JIT team may have a better suggestion or convention we could use here.

Reply:
This handles the following cases
1. mov reg mem
   mov reg mem // identical instructions mem ->reg
2. mov mem reg
   mov mem reg // identical instructions reg ->mem
3. mov mem reg
   mov reg mem  // reg->mem followed by same  mem ->reg
4.  mov reg mem
     mov mem reg // mem->reg followed by same reg-> mem

Reply:
I wanted to note that the back to back moves can have the side effect changing the GC-ness of a location/register.

For example, in this case:
```
//         vmovupd  ymmword ptr[V01 rbp-50H], ymm0  # <-- last instruction
//         vmovupd  ymm0, ymmword ptr[V01 rbp-50H]  # <-- current instruction can be omitted.
```
(Let's pretend vectors can store GC refs, this obviously only applies to GPRs)

If `V01` was a non-GC variable, and `ymm0` in the first instruction held a non-GC value, but then the load was to a GC variable (`ymm0` would become GC-live), we cannot eliminate it.

Does the logic here handle this case already?

Reply:
> Maybe we could call this `IsRedundantStackMov` or similar?
> 
> Someone on the JIT team may have a better suggestion or convention we could use here.

I like this. I think it's descriptive. The reason for keeping it R_S was trying to stick to existing convention but I agree oit might eb a little confusing